### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by oholiab**
Apologies to @smrojas who's ended up duplicating some of this because I never pushed the branch...

I'll test this manually by building a new container from the branch and running it after reviews are in but before merging to master.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import os
+
+with open("README.md", "r") as fh:
+    long_description = fh.read()
+
+setuptools.setup(
+    name="pidtree-bcc",
+    version="0.2",
+    author="Matt Carroll",
+    author_email="mattc@yelp.com",
```
Comment (oholiab) [inline setup.py]: Wasn't quite sure what standard practice was here - entirely happy to change!

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import setuptools
+import os
+
+with open("README.md", "r") as fh:
+    long_description = fh.read()
+
+setuptools.setup(
+    name="pidtree-bcc",
+    version="0.2",
```
Comment (oholiab) [inline setup.py]: Should be 0.3, will fix tomorrow

File: main.py [hunk 3]
```diff
@@ -10,6 +10,7 @@
 from jinja2 import Template
 import contextlib
 from datetime import datetime
+from pidtree_bcc import utils as u
```
Comment (smrojas) [inline main.py]: `utils` isn't a long word. Prefer just using the `utils` name over `u` to avoid ambiguity.

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    author="Matt Carroll",
+    author_email="oholiab@grimmwa.re",
```
Comment (smrojas) [inline setup.py]: mattc@yelp.com ?

File: main.py (hunk 3 above)
Comment (smrojas) [inline main.py]: `utils` isn't a long word. Prefer just using the `utils` name over `u` to avoid ambiguity.

File: tests/test_utils.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
+from pidtree_bcc import utils as u
+import pytest
+import os
+import psutil
```
Comment (smrojas) [inline tests/test_utils.py]: sort

File: tests/test_utils.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
+from pidtree_bcc import utils as u
```
Comment (smrojas) [inline tests/test_utils.py]: put `from`s after `import`s. Ditto about just leaving the name as `utils`

File: tests/test_utils.py [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+@pytest.fixture
+def this_pid():
+    return os.getpid()
+
+def test_crawl_process_tree(this_proc, this_pid):
+    tree = u.crawl_process_tree(this_proc)
+    assert type(tree) is list
+    assert len(tree) >= 1
+    assert tree[0].pid == this_pid
+    assert tree[-1].pid == 1 #should be init
```
Comment (smrojas) [inline tests/test_utils.py]: missing space after `#` in comment

File: .gitignore [hunk 8]
```diff
@@ -1,3 +1,7 @@
 notes.md
 venv
 *~
+.tox/*
+pidtree_bcc.egg-info/*
+*.pyc
+**/__pycache__/*
```
Comment (smrojas) [inline .gitignore]: :nod:

File: setup.py (hunk 1 above)
Comment (smrojas) [reply setup.py]: Probably good as this

File: setup.py (hunk 4 above)
Comment (oholiab) [reply setup.py]: There's more than one commit in this PR, this is the state of the file from an older commit - it's already up to date in the most recent.

File: main.py (hunk 3 above)
Comment (oholiab) [reply main.py]: me: complains about ambiguous naming in code
also me: `import utils as u`

Good call, will fix

File: setup.py (hunk 1 above)
Comment (oholiab) [reply setup.py]: Ack, thanks :)

File: tests/test_utils.py (hunk 7 above)
Comment (oholiab) [reply tests/test_utils.py]: Fixed in 90143be5f4d3710dd3dab6e2f3e800c3f2cc9189

File: tests/test_utils.py (hunk 6 above)
Comment (oholiab) [reply tests/test_utils.py]: Fixed in 90143be5f4d3710dd3dab6e2f3e800c3f2cc9189

File: tests/test_utils.py (hunk 5 above)
Comment (oholiab) [reply tests/test_utils.py]: Fixed in 90143be5f4d3710dd3dab6e2f3e800c3f2cc9189

File: main.py (hunk 3 above)
Comment (oholiab) [reply main.py]: Fixed in 90143be5f4d3710dd3dab6e2f3e800c3f2cc9189

File: setup.py (hunk 2 above)
Comment (oholiab) [reply setup.py]: Fixed in 90143be5f4d3710dd3dab6e2f3e800c3f2cc9189

**Pull request merged**